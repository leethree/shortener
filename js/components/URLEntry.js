/* @flow */

import React from 'react';
import { createFragmentContainer, graphql } from 'react-relay';
import { Card, Row, Column, Link, Pre } from 'rebass';
import { type URLEntry_url } from './__generated__/URLEntry_url.graphql';

type Props = {
  url: URLEntry_url,
};

const URLEntry = ({ url }: Props) => {
  const href = `/${url.id}`;
  return (
    <Card my={8} color="#555555">
      <Row px={8}>
        <Column my={8} flex="1 1 auto">
          {url.url}
        </Column>
        <Column my={8} flex="0 1 auto">
          â†’
        </Column>
        <Column my={8} flex="0 1 auto">
          <Link href={href} target="_blank">
            <Pre>{href}</Pre>
          </Link>
        </Column>
      </Row>
    </Card>
  );
};

export default createFragmentContainer(URLEntry, {
  url: graphql`
    fragment URLEntry_url on URL {
      id
      url
    }
  `,
});
