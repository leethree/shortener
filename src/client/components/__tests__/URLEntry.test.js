import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import URLEntry from '../URLEntry';

test('URLEntry renders', () => {
  const url = {
    id: 'test',
    url: 'https://www.google.com',
  };
  const component = renderer.create(<URLEntry url={url} />);
  expect(component).toMatchSnapshot();
});
