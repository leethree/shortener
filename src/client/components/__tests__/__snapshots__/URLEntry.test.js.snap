// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`URLEntry renders 1`] = `
.c3 {
  box-sizing: border-box;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c8 {
  color: blue;
}

.c6 {
  margin: 0px;
}

.c5 {
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
}

.c9 {
  font-size: 14px;
  font-family: mono;
  margin: 0px;
  overflow: auto;
}

.c1 {
  margin-left: -16px;
  margin-right: -16px;
  padding-left: 512px;
  padding-right: 512px;
}

.c4 {
  margin-bottom: 512px;
  margin-top: 512px;
  padding-left: 16px;
  padding-right: 16px;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
}

.c7 {
  margin-bottom: 512px;
  margin-top: 512px;
  padding-left: 16px;
  padding-right: 16px;
  -webkit-flex: 0 1 auto;
  -ms-flex: 0 1 auto;
  flex: 0 1 auto;
}

.c0 {
  margin-top: 512px;
  margin-bottom: 512px;
  padding: 8px;
  color: #555555;
  background-color: white;
  border-radius: 2px;
  box-shadow: 2;
  overflow: hidden;
}

<div
  className="c0"
>
  <div
    className="c1 c2 c3"
  >
    <div
      className="c4 c5 c6"
      title="https://www.google.com"
    >
      https://www.google.com
    </div>
    <div
      className="c7 c3"
    >
      â†’
    </div>
    <div
      className="c7 c3"
    >
      <a
        className="c8"
        href="/test"
        target="_blank"
      >
        <pre
          className="c9"
        >
          /test
        </pre>
      </a>
    </div>
  </div>
</div>
`;
